site/src/App/Markdown/Markdown.tsx

```diff
@@ -38,69 +34,79 @@ const paragraph = ({ children }: any) => (
-              ? children
-                  .map(({ props }: ReactElement) => props.children)
-                  .join(' ')
-              : children}
-          </LinkableHeading>
-        ) : (
-          <Heading level={resolvedLevel === '5' ? '4' : resolvedLevel}>
-            {children}
-          </Heading>
-        )}
+const renderers: Components = {
+  h1: ({ children }) => (
+    <Box>
+      <Heading level="2">{children}</Heading>
+    </Box>
+  ),
+  h2: ({ children }) => (
+    <Box paddingTop="medium">
+      <LinkableHeading level="3">
+        {Array.isArray(children)
+          ? children.map(({ children: child }) => child).join(' ')
+          : (children as string)}
+      </LinkableHeading>
+    </Box>
+  ),
+  h3: ({ children }) => (
+    <Box paddingBottom="small" paddingTop="medium">
+      <Heading level="4">{children}</Heading>
+    </Box>
+  ),
+  h4: ({ children }) => (
+    <Box paddingBottom="small">
+      <Heading level="4">{children}</Heading>
+    </Box>
+  ),
+  p: paragraph,
+  ol: ({ children }) => (
+    <TextContext.Provider value={null}>
+      <Box paddingBottom="medium">
+        <List space="medium" type="number">
+          {children as ReactNodeNoStrings}
+        </List>
       </Box>
-    );
-  },
-  paragraph,
-  list: ({ children }: { children: any }) => (
+    </TextContext.Provider>
+  ),
+  ul: ({ children }) => (
     <TextContext.Provider value={null}>
       <Box paddingBottom="medium">
-        <List space="medium">{children}</List>
+        <List space="medium">
+          {
+            Children.map(children, (child) =>
+              typeof child === 'string' ? null : child,
+            ) as ReactNodeNoStrings
+          }
+        </List>
       </Box>
     </TextContext.Provider>
   ),
-  listItem: ({ children }: { children: any }) => {
+  li: ({ children }) => {
     const childList = Children.toArray(children);
 
     // @ts-expect-error
-    if (childList[0]?.type?.isParagraph) {
-      return <Stack space="medium">{children}</Stack>;
+    if (childList[1]?.type?.isParagraph) {
+      return <Stack space="medium">{children as ReactNodeNoStrings}</Stack>;
     }
 
     return <Text>{children}</Text>;
   },
-  strong: Strong,
-  emphasis: Strong,
-  inlineCode: InlineCode,
-  code: Code,
-  link: TextLink,
-  linkReference: TextLink,
-  blockquote: ({ children }: any) => (
+  strong: ({ children }) => <Strong>{children}</Strong>,
+  em: ({ children }) => <Strong>{children}</Strong>,
+  pre: ({ children }) => children,
+  code: ({ children, className, ...rest }) => {
+    // This property is added by the `checkInlineCode` rehype plugin.
+    if ('data-inline' in rest && rest['data-inline']) {
+      return <InlineCode>{children as string}</InlineCode>;
+    }
+    const match = /language-(\w+)/.exec(className ?? '');
+    return <Code lang={match?.[1] ?? null} value={children as string} />;
+  },
+  a: ({ children, href }) => (
+    <TextLink href={href ?? '#!'}>{children}</TextLink>
```

Comment:
Is `#!` a special href of some kind? I've never seen it anywhere. Would typically use just `#` for links that intentionally don't go anywhere.
```suggestion
    <TextLink href={href ?? '#!'}>{children}</TextLink>
```

Reply:
`#` goes to the top of the page, `#!` acts as an nonexistent anchor and doesn't go anywhere (doesn't scroll to page top)